transport/pool.go

```diff
@@ -160,6 +176,23 @@ func (r *PoolRefiller) onConnClose(conn *Conn) {
 	}
 }
 
+func validateKeyspace(keyspace string) error {
+	if keyspace == "" || len(keyspace) > 48 {
+		return fmt.Errorf("keyspace: invalid length")
+	}
+
+	for _, c := range keyspace {
+		if !(unicode.IsLetter(c) || unicode.IsDigit(c) || c == '_') {
+			return fmt.Errorf("keyspace: illegal characters present")
+		}
+	}
+	return nil
+}
+
+func isIOError(err error) bool {
+	return strings.HasPrefix(err.Error(), "IO")
+}
+
```

Comment:
Ha Ha Ha

https://pkg.go.dev/errors - error wrapping and unwrapping
https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/net.go;l=439 - the I/O errors, look for error types